pkg/utils/grpc.go

```diff
@@ -34,37 +59,124 @@ func NewGrpc(serverType string, port int, certPem, keyPem, rootCertPem []byte) (
 		}),
 	}
 
-	mutualTLS, err := setupMutualTLS(false, serverType, certPem, keyPem, rootCertPem)
-	if err != nil {
-		log.Error().Err(err).Msg("Error setting up mutual tls for GRPC server")
-		return nil, nil, err
+	// #nosec G402: TLS MinVersion too low
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: false,
+		ServerName:         s.name,
+		ClientAuth:         tls.RequireAndVerifyClientCert,
+		MinVersion:         constants.MinTLSVersion,
+		GetConfigForClient: func(*tls.ClientHelloInfo) (*tls.Config, error) {
+			// use lock to prevent concurrent updates and reads to the tls config
+			s.mu.Lock()
+			defer s.mu.Unlock()
+			return &s.config, nil
+		},
 	}
+	mutualTLS := grpc.Creds(credentials.NewTLS(&tlsConfig))
 	grpcOptions = append(grpcOptions, mutualTLS)
 
-	return grpc.NewServer(grpcOptions...), lis, nil
+	s.server = grpc.NewServer(grpcOptions...)
+	return s, lis, nil
+}
+
+// GetServer returns the gRPC server
+func (s *Server) GetServer() *grpc.Server {
+	return s.server
+}
+
+func (s *Server) updateConfig() error {
+	grpcCert, err := s.cm.IssueCertificate(
+		s.certCommonName,
+		certificate.Internal,
+		certificate.FullCNProvided())
+	if err != nil {
+		return err
+	}
+
+	certif, err := tls.X509KeyPair(grpcCert.GetCertificateChain(), grpcCert.GetPrivateKey())
+	if err != nil {
+		return fmt.Errorf("failed loading Certificate (%+v) and Key (%+v) PEM files", grpcCert.GetCertificateChain(), grpcCert.GetPrivateKey())
+	}
+
+	certPool := x509.NewCertPool()
+
+	// Load the set of Root CAs
+	if ok := certPool.AppendCertsFromPEM(grpcCert.GetTrustedCAs()); !ok {
+		return fmt.Errorf("failed to append client certs")
+	}
+
+	// use lock to prevent concurrent updates and reads to the tls config
+	s.mu.Lock()
+	// #nosec G402: TLS MinVersion too low
```

Comment:
You should theoretically be able to remove this since non-FIPS builds have a MinTLSVersion of 1.3

Reply:
I was getting a lint error without this. I'll try removing it again